lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -205,6 +266,74 @@ export default function SubmitGradeForm({
             )}
           </span>
 
+          {acceptGradingComments && (
+            <span className="relative">
+              <Button
+                icon={commentIsSet ? NoteFilledIcon : NoteIcon}
+                disabled={disabled}
+                title={commentIsSet ? 'Edit comment' : 'Add comment'}
+                onClick={() => setShowCommentControls(prev => !prev)}
+                classes={classnames(
+                  'border border-r-0 rounded-none ring-inset h-full relative',
+                  'disabled:opacity-50'
+                )}
+              />
+              {showCommentControls && (
+                <div
```

Comment:
I agree on this, but I think I will do that on a follow-up PR.

I was exploring this, and there are a couple of grey areas where we need to make some decisions. I don't want that to become a distraction, so I think it will be helpful to address it as a thing of its own.